### PR TITLE
Fix Migration Batching When Seed Data Has Non-Writable Columns (I.E. RowVersion)

### LINKED ISSUE #12466: Don't use Min/Max batch size values when generating migrations
I've noticed that sometimes when I'm inserting multiple entities of the same type with hasdata they will each get their own insertdata command. Other times I've seen it use one insertdata command and list multiple items in the objects parameter. I haven't been able to track down the pattern for what results in one or the other. 

Here is a screenshot of a migration wher eyou can see that brewertypes are combined into one insertdata command but down below there are two separate commands for inserting 2 Units (the 2nd one is chopped off in the screenshot).
image

One difference between BrewerType and Unit is that BrewerType owns an another type via OwnsOne --which is supplying some of the values for each of those rows. Is that enough to make the difference?

Figured I would ask rather than continue to try and try.

Thanks!

### Further technical details
EF Core version: 2.1.1
Database Provider:  Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE:  Visual Studio 2017 15.7.3

### PR DESCRIPTION
**Pull request by crowet**
Please check if the PR fulfills these requirements

- [x] The code builds and tests pass (verified by our automated build checks)
- [x] Commit messages follow this format
        Regression from pull request #13510
        Fix migration scripts when seed data has non-writable columns (i.e. rowversion)
        Prior to regression, migration scripts would insert seed data in batching.
        This fix restores behavior.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code meets the expectations our engineering guidelines.
      https://github.com/aspnet/AspNetCore/wiki/Engineering-guidelines

Review the guidelines for CONTRIBUTING.md for more details.

### CODE REVIEW

**Comment by smitpatel** [comment]
Remove all the processing around batch size. It does not fix #12466. This issue only stopped batching when the columns contained non-writable case not other cases. #12466 does not have non-writable column.

**Comment by crowet** [comment]
@smitpatel I removed the issue reference. This is certainly an issue, but I'm not sure if it has been reported before.

File: test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs [hunk 1]
```diff
@@ -1374,6 +1375,92 @@ public void Add_column_with_seed_data()
...
+                            v => Assert.Equal(44, v));
+                    },
+                    o =>
+                    {
+                        var m = Assert.IsType<DeleteDataOperation>(o);
+                        AssertMultidimensionalArray(
+                            m.KeyValues,
+                            v => Assert.Equal(45, v));
+                    }),
+                builderOptions => builderOptions.UseFakeRelational(a => a.MaxBatchSize(4)));
```
Comment (smitpatel) [inline test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs]: Do you really need this option? It should work without it.

Comment (crowet) [reply test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs]: The default batch size in the Relational abstract class is 1. This fails if the batch size setting not set. The test requires all 4 inserts included in the same batch, but when that option is removed, it becomes 4 operations.

image

**Comment by AndriySvyryd** [comment]
@crowet Thanks for your contribution!

**Pull request merged**